### PR TITLE
Refactor all the things

### PR DESCRIPTION
**Pull request by schrej**
The primary goal of this refactoring is to reduce the tight coupling between the packages that exists right now. This should greatly ease testing, in case we ever get to that.

I'll try to list everything I did, but it's still WIP.

- new API client that has to be instantiated, can then be replaced by a mocked client for testing
- new server.Manager instead of server.GetServers() singleton

### CODE REVIEW

File: panelapi/util.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import (
+	"net/http"
+
+	"github.com/apex/log"
+)
+
+// Logs the request into the debug log with all of the important request bits.
+// The authorization key will be cleaned up before being output.
+//
+// TODO(schrej): Somehow only execute the logic when log level is debug.
```
Comment (DaneEveritt) [inline panelapi/util.go]: `c.Debug` in the config is the thing that sets debug on the logger.

File: router/middleware.go [hunk 2]
```diff
@@ -91,14 +94,19 @@ func (m *Middleware) RequireAuthorization() gin.HandlerFunc {
...
-	return server.GetServers().Find(func(s *server.Server) bool {
-		return uuid == s.Id()
-	})
+func (m *Middleware) WithServerManager() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		c.Set("servermanager", m.serverManager)
+	}
+}
+
+func ServerManagerFromContext(c *gin.Context) server.Manager {
```
Comment (DaneEveritt) [inline router/middleware.go]: I've been using `Extract___` as the naming for these guys, e.g. `ExtractServerManager()`

File: panelapi/util.go (hunk 1 above)
Comment (schrej) [reply panelapi/util.go]: yeah, but panelapi shouldn't import the config

File: router/middleware.go (hunk 2 above)
Comment (schrej) [reply router/middleware.go]: Yeah I saw, I guess I'll rename it

File: panelapi/util.go (hunk 1 above)
Comment (DaneEveritt) [reply panelapi/util.go]: Make it just have a debug flag passed in? Does it really matter much in the end though, its hardly a complex function.

File: sftp/sftp.go [hunk 3]
```diff
@@ -42,33 +41,25 @@ func Initialize(config config.SystemConfiguration) error {
...
 	}
-
-	return s.Filesystem().HasSpaceAvailable(true)
+	return srv.Filesystem().HasSpaceAvailable(true)
 }
 
 // Validates a set of credentials for a SFTP login against Pterodactyl Panel and returns
 // the server's UUID if the credentials were valid.
-func validateCredentials(c api.SftpAuthRequest) (*api.SftpAuthResponse, error) {
+func (s *Server) validateCredentials(c api.SftpAuthRequest) (*api.SftpAuthResponse, error) {
```
Comment (DaneEveritt) [inline sftp/sftp.go:62]: Why is this part of the `Server` struct if it does nothing with said server?

**Review by DaneEveritt — APPROVED** [review]

**Review by DaneEveritt — CHANGES_REQUESTED** [review]
hold

**Pull request merged**